utils/emmylua/wesnoth/units.lua

```diff
@@ -0,0 +1,265 @@
+---@meta
+
+---Defines a weapon possessed by a unit
+---@class unit_weapon
+---@field read_only boolean
+---@field description tstring
+---@field name string
```

Comment:
tstring ?

Reply:
No, this is the name key from WML, so it's functionally the weapon ID.

Reply:
Thanks.